homeassistant/components/stream/hls.py

```diff
@@ -187,28 +321,91 @@ class HlsSegmentView(StreamView):
+        # If these conditions aren't met then we return a 416.
+        # http_range_start can be None, so use a copy that uses 0 instead of None
+        if (http_start := request.http_range.start or 0) > segment.data_size or (
+            segment.complete and http_start >= segment.data_size
+        ):
+            return web.HTTPRequestRangeNotSatisfiable(
+                headers={
+                    "Cache-Control": f"max-age={track.target_duration:.0f}",
+                    "Content-Range": f"bytes */{segment.data_size}",
+                }
+            )
+        headers = {
+            "Content-Type": "video/iso.segment",
+            "Cache-Control": f"max-age={6*track.target_duration:.0f}",
+        }
+        # For most cases we have a 206 partial content response.
+        status = 206
+        # For the 206 responses we need to set a Content-Range header
+        # See https://datatracker.ietf.org/doc/html/rfc8673#section-2
+        if request.http_range.stop is None:
+            if request.http_range.start is None:
+                status = 200
+                if segment.complete:
+                    # This is a request for a full segment which is already complete
+                    # We should return a standard 200 response.
+                    return web.Response(
+                        body=segment.get_data(), headers=headers, status=status
+                    )
+                # Otherwise we still return a 200 response, but it is aggregating
+                http_stop = float("inf")
+            else:
+                # See https://datatracker.ietf.org/doc/html/rfc7233#section-2.1
+                headers[
+                    "Content-Range"
+                ] = f"bytes {http_start}-{(http_stop:=segment.data_size)-1}/*"
+        else:  # The remaining cases are all 206 responses
+            if segment.complete:
+                # If the segment is complete we have total size
+                headers["Content-Range"] = (
+                    f"bytes {http_start}-"
+                    + str(min(request.http_range.stop, segment.data_size) - 1)
+                    + f"/{segment.data_size}"
+                )
+            else:
+                # If we don't have the total size we use a *
+                headers[
+                    "Content-Range"
+                ] = f"bytes {http_start}-{request.http_range.stop-1}/*"
+        # Set up streaming response that we can write to as data becomes available
+        response = web.StreamResponse(headers=headers, status=status)
+        # Waiting until we write to prepare *might* give clients more accurate TTFB
+        # and ABR measurements, but it is probably not very useful for us since we
+        # only have one rendition anyway. Just prepare here for now.
+        await response.prepare(request)
+        try:
+            for bytes_to_write in segment.get_aggregating_bytes(
+                start_loc=http_start, end_loc=request.http_range.stop or http_stop
```

Comment:
One suggestion could be to always assign `http_stop` as a variable and use it above when formatting the content rage value, then reuse it here.  I realize this function probably handles the case where `request.http_range.stop` is greater than `segment.data_size` however, by assigning it to the min above, for example, it could remove that as a concern.

Reply:
Ok, let's do that and see if it's clearer.